src/transaction/log_recovery_redo.hpp

```diff
@@ -644,6 +645,13 @@ void log_rv_redo_record_sync_apply (THREAD_ENTRY *thread_p, log_rv_redo_context
 			 VPID_AS_ARGS (&rcv_vpid), (int) rcv.offset, (int) rcv.length);
     }
 
+  if (is_passive_transaction_server() && log_data.rcvindex == RVPGBUF_DEALLOC)
```

Comment:
How about process this in pgbuf_rv_dealloc_redo()?

Reply:
In pgbuf_rv_dealloc_redo (), we can not modify the rcv.pgptr value because the function receives that value as a constant. 
And the format of the recovery functions is predetermined, so it cannot be changed. This is because it is a member of the `rvfun`